src/runtime/virtcontainers/hypervisor.go

```diff
@@ -705,6 +705,20 @@ func (conf *HypervisorConfig) assetPath(t types.AssetType) (string, error) {
 		return a.Path(), nil
 	}
 
+	if t == types.ImageAsset {
+		_, ok = conf.customAssets[types.InitrdAsset]
+		if ok {
+			// Use the initrd custom asset instead of the configured image.
+			return "", nil
+		}
+	} else if t == types.InitrdAsset {
+		_, ok = conf.customAssets[types.ImageAsset]
+		if ok {
+			// Use the image custom asset instead of the configured initrd.
+			return "", nil
+		}
+	}
```

Comment:
It is a bit confusing to return an initrd path for `ImageAsset`or vice versa.

Reply:
Just for the record, this code returns:

1. An empty path for ImageAsset when it wants to force the caller to use a custom InitrdAsset.
2. An empty path for InitrdAsset when it wants to force the caller to use a custom ImageAsset.

It doesn't return an initrd path for ImageAsset or vice versa.

With that said, I agree that https://github.com/kata-containers/kata-containers/pull/7716 is a cleaner solution.

Thanks!